src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/enumeration/ReferenceDataType.kt

```diff
@@ -34,5 +35,10 @@ enum class ReferenceDataType(val domain: String) {
 
     fun fromDomain(domain: String) =
       map[domain] ?: throw InvalidDomainException("Fail to map $domain to Reference Data Type. $VALIDATION_DESCRIPTION")
+
+    fun getOutcomeType(code: String, referenceDataRepository: ReferenceDataRepository) =
```

Comment:
This is a good idea although it's really a repo function and probably an extension method. Maybe try
```
fun ReferenceDataRepository.getOutcomeType(code: String) =
      findByDomainAndCode(OUTCOME_TYPE, code)?.also {
        require(it.isActive()) { "OUTCOME_TYPE code '$code' is inactive" }
      } ?: throw IllegalArgumentException("OUTCOME_TYPE code '$code' does not exist")
```